lib/uploadcare/rails/active_record/mount_uploadcare_file.rb

```diff
@@ -1,72 +1,89 @@
 # frozen_string_literal: true
 
+require 'active_record'
 require 'active_support/concern'
+require 'uploadcare/errors/mount_error'
 
 module Uploadcare
   module Rails
-    # A module containing ActiveRecord extension. Allows to use uploadcare file methods in Rails models
     module ActiveRecord
-      extend ActiveSupport::Concern
+      # A module containing ActiveRecord extension. Allows to use uploadcare file methods in Rails models
+      module MountUploadcareFile
+        extend ActiveSupport::Concern
 
-      UUID_REGEX = /\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b/.freeze
+        FILE_ID_REGEX = /\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b/.freeze
 
-      def build_uploadcare_file(attribute)
-        cdn_url = attributes[attribute.to_s].to_s
-        return nil if cdn_url.empty?
+        def build_uploadcare_file(attribute)
+          cdn_url = attributes[attribute.to_s].to_s
+          return nil if cdn_url.empty?
 
-        uuid = cdn_url.match(UUID_REGEX).to_s
-        ::Rails.cache.read(cdn_url) || Uploadcare::Rails::File.new(cdn_url: cdn_url, uuid: uuid.presence)
-      end
+          uuid = cdn_url.match(FILE_ID_REGEX).to_s.presence
+          ::Rails.cache.read(cdn_url) || Uploadcare::Rails::File.new(cdn_url: cdn_url, uuid: uuid)
+        end
 
-      def log_uploadcare_error(exception, message)
-        logger.error message
-        logger.error ::Rails.backtrace_cleaner.clean(exception.backtrace).join("\n ").to_s
-      end
+        def log_uploadcare_error(exception, message)
+          logger.error message
+          logger.error ::Rails.backtrace_cleaner.clean(exception.backtrace).join("\n ").to_s
+        end
 
-      class_methods do
-        # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Lint/NestedMethodDefinition
-        def mount_uploadcare_file(attribute, _options = {})
-          define_method attribute.to_s do
-            build_uploadcare_file attribute
-          end
+        class_methods do
+          # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Lint/NestedMethodDefinition
+          def mount_uploadcare_file(attribute, _options = {})
+            validate_file_mount!(attribute)
 
-          define_method "store_#{attribute}!" do
-            file = build_uploadcare_file attribute
-            return unless file
+            define_singleton_method "has_uploadcare_file_for_#{attribute}?" do
```

Comment:
Why it is always true?

Reply:
because when we mount `uploadcare file` to a model, this model always has an `uploadcare_file`